### PR TITLE
fix(events_targets): installing latest aws sdk fails in cn partition

### PR DESCRIPTION
**Pull request by GavinZZ**
### Issue # (if applicable)

Closes https://github.com/aws/aws-cdk/issues/29373

### Reason for this change

AWS Log Group event target by default installs the latest aws sdk for its custom resource and this would fail in `aws-cn` partition. This PR exposes the `installLatestAwsSdk` to the surface and allows users to optionally turn off `installLatestAwsSdk` for cloudwatch log events target.

### Description of changes

Allow users to override the value, if unset default to true which is the same behaviour as current.

### Description of how you validated changes

all tests pass.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-events-targets/lib/log-group-resource-policy.ts [hunk 1]
```diff
@@ -39,6 +43,7 @@ export class LogGroupResourcePolicy extends cr.AwsCustomResource {
         },
         physicalResourceId: cr.PhysicalResourceId.of(id),
       },
+      installLatestAwsSdk: props.installLatestAwsSdk,
```
[the remark this answers is not in the document]
Comment (pahud) [reply packages/aws-cdk-lib/aws-events-targets/lib/log-group-resource-policy.ts:46]: super() essentially instantiate the aws.CustomResource and the `installLatestAwsSdk` prop passes all the why to its props.

https://github.com/aws/aws-cdk/blob/4ee050fb081b591be53175eeda2180e22fe8aa02/packages/aws-cdk-lib/aws-events-targets/lib/log-group-resource-policy.ts#L46

[the remark this answers is not in the document]
Comment (GavinZZ) [reply packages/aws-cdk-lib/aws-events-targets/lib/log-group-resource-policy.ts:46]: When this property is either unset or set to `true`, it's exactly the same behaviour as current. It basically passes the value to inside `super()` call when creating the custom resource [here](https://github.com/aws/aws-cdk/blob/6c9063b027a5a1599126bc9f3f154d1b9a57b6ed/packages/aws-cdk-lib/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts#L499).

File: packages/aws-cdk-lib/aws-events-targets/lib/log-group-resource-policy.ts [hunk 2]
```diff
@@ -15,6 +15,10 @@ export interface LogGroupResourcePolicyProps {
    * The policy statements for the log group resource logs
    */
   readonly policyStatements: [iam.PolicyStatement];
+  /**
+   * Whether to install latest AWS SDK for the custom resource
+   */
+  readonly installLatestAwsSdk?: boolean;
```
Comment (pahud) [inline packages/aws-cdk-lib/aws-events-targets/lib/log-group-resource-policy.ts:21]:
```suggestion
  /**
   * Whether to install latest AWS SDK for the custom resource
   * @default - install latest AWS SDK
   */
  readonly installLatestAwsSdk?: boolean;
```

I guess we should always have a @default tag for optional prop like this.

**Review by pahud — CHANGES_REQUESTED** [review]

File: packages/aws-cdk-lib/aws-events-targets/lib/log-group-resource-policy.ts (hunk 2 above)
Comment (GavinZZ) [reply packages/aws-cdk-lib/aws-events-targets/lib/log-group-resource-policy.ts:21]: Good point, will update it.

File: packages/aws-cdk-lib/aws-events-targets/lib/log-group-resource-policy.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulhcsun) [reply packages/aws-cdk-lib/aws-events-targets/lib/log-group-resource-policy.ts:48]: Oh cool, thanks for clarifying!

**Review by paulhcsun — APPROVED** [review]

**Pull request merged**